strapi/strapi
# enhancement!: use server.globalProxy value for creating a global http/https proxy

### What does it do?

- Move `server.proxy` to `server.proxy.koa`
- Move `server.globalProxy` to `server.proxy.global`
- add `server.proxy.fetch` to act only on Fetch requests  (defaults to proxy.global if set)
- add `server.proxy.http` to act on `http` requests (defaults to proxy.global if set)
- add `server.proxy.https` to act on `https` requests (defaults to proxy.global if set)

This

### Why is it needed?

Allows proxy of all http and https requests globally (including anything that uses http/https internally, such as axios, passport-oauth2, etc, giving Strapi an actual global proxy to solve issues like [proxied SSO](https://github.com/strapi/strapi/issues/17317)

### How to test it?

- set server.proxy.global (or ad individual value) in your config
- use something like `npx http-echo-server` to create an echo proxy
- set up SSO (or whatever feature you want to test proxied)
- when you use SSO, you should see it proxied through your echo server
- fetch requests (such as telemetry) should still come through to the echo proxy

### Related issue(s)/PR(s)

Fixes https://github.com/strapi/strapi/issues/17317

DX-989

Documentation PR: https://github.com/strapi/documentation/pull/2124

TODO:

- [ ] write a codemod to migrate server.proxy -> server.proxy.koa and server.globalProxy -> server.proxy.global which will come in a later PR
